sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestProxyTestBase.java

```diff
@@ -0,0 +1,509 @@
+    private AzureProfile testProfile;
+    private AuthFile testAuthFile;
+    private boolean isSkipInPlayback;
+
+    /**
+     * Sets upper bound execution timeout for each @Test method.
+     * {@link org.junit.jupiter.api.Timeout} annotation on test methods will only narrow the timeout, not affecting the upper
+     * bound.
+     */
+    @RegisterExtension
+    final PlaybackTimeoutInterceptor playbackTimeoutInterceptor = new PlaybackTimeoutInterceptor(() -> Duration.ofSeconds(30));
+
+    /**
+     * Generates a random resource name.
+     *
+     * @param prefix Prefix for the resource name.
+     * @param maxLen Maximum length of the resource name.
+     * @return A randomly generated resource name with a given prefix and maximum length.
+     */
+    protected String generateRandomResourceName(String prefix, int maxLen) {
+        return testResourceNamer.randomName(prefix, maxLen);
+    }
+
+    /**
+     * @return A randomly generated UUID.
+     */
+    protected String generateRandomUuid() {
+        return testResourceNamer.randomUuid();
+    }
+
+    /**
+     * @return random password
+     */
+    public static String password() {
+        // do not record
+        String password = new ResourceNamer("").randomName("Pa5$", 12);
+        LOGGER.info("Password: {}", password);
+        return password;
+    }
+
+    private static String sshPublicKey;
+
+    /**
+     * @return an SSH public key
+     */
+    public static String sshPublicKey() {
+        if (sshPublicKey == null) {
+            try {
+                KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+                keyGen.initialize(1024);
+                KeyPair pair = keyGen.generateKeyPair();
+                PublicKey publicKey = pair.getPublic();
+
+                RSAPublicKey rsaPublicKey = (RSAPublicKey) publicKey;
+                ByteArrayOutputStream byteOs = new ByteArrayOutputStream();
+                DataOutputStream dos = new DataOutputStream(byteOs);
+                dos.writeInt("ssh-rsa".getBytes(StandardCharsets.US_ASCII).length);
+                dos.write("ssh-rsa".getBytes(StandardCharsets.US_ASCII));
+                dos.writeInt(rsaPublicKey.getPublicExponent().toByteArray().length);
+                dos.write(rsaPublicKey.getPublicExponent().toByteArray());
+                dos.writeInt(rsaPublicKey.getModulus().toByteArray().length);
+                dos.write(rsaPublicKey.getModulus().toByteArray());
+                String publicKeyEncoded = new String(Base64.getEncoder().encode(byteOs.toByteArray()), StandardCharsets.US_ASCII);
+                sshPublicKey = "ssh-rsa " + publicKeyEncoded;
+            } catch (NoSuchAlgorithmException | IOException e) {
+                throw LOGGER.logExceptionAsError(new IllegalStateException("failed to generate ssh key", e));
+            }
+        }
+        return sshPublicKey;
+    }
```

Comment:
I assume these are duplicate code? Either use a common base, or move them to util class.

Reply:
I am fine with some duplicate code, but not lots of them.

Reply:
Currently ResourceManagerTestProxyTestBase is just a copy of ResourceManagerTestBase, with updates in 'code change' section in description to fit in TestProxy.
![image](https://github.com/Azure/azure-sdk-for-java/assets/87355844/159f49f3-1b04-483f-b6e1-2285271a2c75)

Let me try to refine the code to make it easier to reuse the code in ResourceManagerTestBase and easier to see the difference.